K2Bridge/Factories/AggregateFactory.cs

```diff
@@ -56,21 +59,29 @@ public static BucketAggregate GetDateHistogramAggregate(string key, DataTable da
         /// <param name="dataTable">The row collection be parsed.</param>
         /// <param name="logger">ILogger object for logging.</param>
         /// <returns><see cref="BucketAggregate"/>.</returns>
-        public static BucketAggregate GetRangeAggregate(string key, DataTable dataTable, ILogger logger)
+        public static BucketAggregate GetRangeAggregate(string key, DataTable dataTable, DataTable metadataTable, ILogger logger)
         {
+            Ensure.IsNotNull(metadataTable, nameof(DataTable));
+
             logger.LogTrace("Get range aggregate for {}", key);
 
             var rangeAggregate = new BucketAggregate() { Keyed = true };
 
+            var outputBuckets = new HashSet<string>();
+
             foreach (DataRow row in dataTable.Rows)
             {
                 var bucket = BucketFactory.CreateRangeBucket(key, row, logger);
                 if (bucket != null)
                 {
                     rangeAggregate.Buckets.Add(bucket);
+                    outputBuckets.Add(Convert.ToString(row[key]));
                 }
             }
 
+            // Add missing buckets
+            AddMissingBuckets(key, dataTable, metadataTable, outputBuckets, rangeAggregate, BucketFactory.CreateRangeBucket, logger);
```

Comment:
Personally, I prefer to keep directly the foreach done by AddMissingBuckets inside the Aggregate and avoid the delegate usage. OK, it will add a couple of lines but will be more readable.

Reply:
Yeah, it's a bit heavy 😬 We might have to use it also for Date Range at some point. I'm happy both ways :) @AsafMah

Reply:
I'm really torn on this too, 7 parameters is a lot.
For now it seems that we don't need it in date range, right? (they work fine with empty buckets).
So I think maybe go back to duplicating the lines, and if we'll need it 3 times we might want to do a more serious refactor (maybe put it into data types instead of static functions).

But your choice really.

Reply:
OK, I moved back to duplicating the code, and we can decide in the future to refactor.